dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -503,7 +503,7 @@ class ScriptLevelsControllerTest < ActionController::TestCase
 
   test "should 404 for invalid script level for twenty hour" do
     assert_raises(ActiveRecord::RecordNotFound) do # renders a 404 in prod
-      get :show, script_id: Script::TWENTY_HOUR_ID, id: 40000
+      get :show, script_id: Script::TWENTY_HOUR_NAME, id: 40000
```

Comment:
yep, see script_levels_controller.rb line 25. it calls the same thing that ScriptsController does (Script.get_from_cache)

The router itself does not check the 'type' of the script_id, it takes any string (including strings that happen to be integers) and passes it to the controller